rectools/visuals/visual_app.py

```diff
@@ -268,239 +455,349 @@ def display(self) -> None:
-    auto_display : bool, optional, default ``True``
-        Display widgets right after initialization.
-    formatters : tp.Optional[tp.Dict[str, tp.Callable]], optional, default ``None``
-        Formatter functions to apply to columns elements in the sections of interactions and recos.
-        Keys of the dict must be columns names (item_data, interactions and recos columns can be
-        specified here). Values bust be functions that will be applied to corresponding columns
-        elements. The result of each function must be a unicode string that represents html code.
-        Formatters can be used to format text, create links and display images with html.
-    rows_limit : int, optional, default 20
-        Maximum number of rows to display in the sections of interactions and recos.
-    min_width : int, optional, default 100
-        Minimum column width in pixels for dataframe columns in widgets output. Must be greater then
-        10.
-
-    Examples
-    --------
-    >>> recos = {
-    ...     "model1": pd.DataFrame({Columns.User: [1, 2], Columns.Item: [3, 4], Columns.Score: [0.99, 0.9]}),
-    ...     "model2": pd.DataFrame({Columns.User: [1, 2], Columns.Item: [5, 6], Columns.Rank: [1, 1]})
-    ... }
-    >>>
-    >>> item_data = pd.DataFrame({
-    ...     Columns.Item: [3, 4, 5, 6, 7, 8],
-    ...     "feature_1": ["one", "two", "three", "five", "one", "two"]
-    ... })
-    >>>
-    >>> interactions = pd.DataFrame({Columns.User: [1, 1, 2], Columns.Item: [3, 7, 8]})
-    >>> selected_users = {"user_one": 1}
-    >>> formatters = {"item_id": lambda x: f"<b>{x}</b>"}
-    >>>
-    >>> widgets = VisualApp(
-    ...     recos=recos,
-    ...     item_data=item_data,
-    ...     interactions=interactions,
-    ...     selected_users=selected_users,
-    ...     formatters=formatters,
-    ...     auto_display=False
-    ... )
-    """
+        Parameters
+        ----------
+        folder_name : str
+            Destination folder for data.
+        overwrite : bool, default ``False``
+            Allow to overwrite in the folder files if they already exist.
+        """
+        self.data_storage.save(folder_name, overwrite)
 
-    def __init__(
-        self,
-        recos: TablesDict,
-        interactions: pd.DataFrame,
-        item_data: pd.DataFrame,
-        selected_users: tp.Dict[tp.Hashable, tp.Hashable],
-        n_random_users: int = 0,
+    @classmethod
+    def load(
+        cls,
+        folder_name: str,
         auto_display: bool = True,
         formatters: tp.Optional[tp.Dict[str, tp.Callable]] = None,
         rows_limit: int = 20,
         min_width: int = 100,
-    ) -> None:
-        super().__init__(
-            recos=recos,
-            interactions=interactions,
-            item_data=item_data,
-            selected_users=selected_users,
-            n_random_users=n_random_users,
+    ) -> "VisualAppBase":
```

Comment:
Incorrect signature in case you have child classes

Reply:
fixed